### PR TITLE
feat: support handlebars layout

### PR DESCRIPTION
**Pull request by sinchang**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

support handlebars layout

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -18,8 +20,18 @@ module.exports = app => {
...
-      return this[COMPILE](content, context, options);
+      const viewContent = await fs.readFile(name, 'utf8');
+      const body = await this[COMPILE](viewContent, context, options);
+      let layout = context.layout;
+
+      if (layout === false || !defaultLayout) return body;
+
+      layout = layout ? layout : defaultLayout;
+
+      if (fs.existsSync(layoutsPath)) {
```
Comment (atian25) [inline app.js]: `await fs.exist()`

File: app.js [hunk 2]
```diff
@@ -18,8 +20,18 @@ module.exports = app => {
...
+        const layoutContent = fs.readFileSync(`${layoutsPath}/${layout}.hbs`).toString();
```
Comment (atian25) [inline app.js]: `const layoutContent = await fs.readFile(path.join(layoutsPath, `${layout}.hbs`), 'utf-8');`

**Review by atian25 — APPROVED** [review]
@popomore 看看

File: app.js [hunk 3]
```diff
@@ -18,8 +20,20 @@ module.exports = app => {
...
-      return this[COMPILE](content, context, options);
+      const viewContent = await fs.readFile(name, 'utf8');
+      const body = await this[COMPILE](viewContent, context, options);
+      let layout = context.layout;
+
+      if (layout === false || !defaultLayout) return body;
+
+      layout = layout ? layout : defaultLayout;
+
+      const isExistLayoutsPath = await fs.exists(layoutsPath);
```
Comment (popomore) [inline app.js:31]: 这个检查可以放到 app.js 里面去做，不然每次渲染都要做。

File: app.js [hunk 4]
```diff
@@ -18,8 +20,20 @@ module.exports = app => {
     }
 
     async render(name, context, options) {
-      const content = await fs.readFile(name, 'utf8');
-      return this[COMPILE](content, context, options);
+      const viewContent = await fs.readFile(name, 'utf8');
+      const body = await this[COMPILE](viewContent, context, options);
+      let layout = context.layout;
```
Comment (popomore) [inline app.js:25]: 这个参数最好从 options 读

File: app.js [hunk 5]
```diff
@@ -18,8 +20,20 @@ module.exports = app => {
...
+
+      if (layout === false || !defaultLayout) return body;
+
+      layout = layout ? layout : defaultLayout;
+
+      const isExistLayoutsPath = await fs.exists(layoutsPath);
+
+      if (isExistLayoutsPath) {
+        const layoutContent = await fs.readFile(path.join(layoutsPath,`${layout}.hbs`), 'utf-8');
```
Comment (popomore) [inline app.js:34]: 加个读文件缓存

**Review by popomore — CHANGES_REQUESTED** [review]

**Pull request closed without merging**